### ISSUE TITLE
problem with OFFSET and FETCH when the order is by a column that has one value

### ISSUE THREAD

**Issue by luizfbicalho** [aspnet/EntityFramework6#162]
If I have a class like this for example

`class Entity{
    int Id {get;set;}
    string Description {get;set;}
    int Size {get;set;} // size can be 1,2 or 3
}`

and I query this data in a grid with order and pagins ( in my case I'm using kendo ui) and I order by size, and filter where the size == 1, I have problems when I change the pages, the data that is returned is the same every page.

the sql is like this

SELECT * 
FROM Entity
WHERE Size = 1
ORDER BY Size
OFFSET 10 ROWS 
FETCH NEXT 10 ROWS ONLY

and to solve this problem it should be like 

SELECT * 
FROM Entity
WHERE Size = 1
ROW_NUMBER() over (ORDER BY Size ASC) 
OFFSET 10 ROWS 
FETCH NEXT 10 ROWS ONLY

this is caused by this feature, if it's not possible to change the order, ate least I need a way to change the sql genereator.

https://data.uservoice.com/forums/72025-entity-framework-feature-suggestions/suggestions/2541081-use-offset-and-fetch-for-skip-and-take-in-sql-serv

**Comment by ErikEJ** [comment]
You can force a 2008 manifest token http://stackoverflow.com/questions/4741499/how-to-configure-providermanifesttoken-for-ef-code-first

**Comment by maumar** [comment]
@luizfbicalho could you provide some more details about your scenario (model used, listing of sql that is being issues, database used etc). I tried the following:

```cs
        static void Main(string[] args)
        {
            Database.SetInitializer(new DropCreateDatabaseAlways<MyContext>());
            using (var ctx = new MyContext())
            {
                for (int i = 0; i < 50; i++)
                {
                    var e = new MyEntity { Description = "d" + i, Size = 1 };
                    ctx.Entities.Add(e);
                }

                ctx.SaveChanges();
            }

            using (var ctx = new MyContext())
            {
                ctx.Database.Log = Console.WriteLine;
                var batch1 =  ctx.Entities.OrderBy(e => e.Size).Where(e => e.Size == 1).Skip(10).Take(10).ToList();
                var batch2 = ctx.Entities.OrderBy(e => e.Size).Where(e => e.Size == 1).Skip(20).Take(10).ToList();
            }
        }
```

and while SQL looks non-deterministic, the results I'm getting back are consistently correct (i.e. rows 11-20 for first batch and 21-30 for second batch). I guess the data gets returned in the order that is internally stored on the server. I wonder if there is anything different about your scenario that results in different behavior.

**Comment by luizfbicalho** [comment]
@maumar  thanks for looking at this problem, I used your example and it didn't make the error, so I changed it a little

`namespace EFError
{
    public class Program
    {
        static void Main(string[] args)
        {
            Database.SetInitializer(new DropCreateDatabaseAlways<MyContext>());
            using (var ctx = new MyContext())
            {
                for (int i = 0; i < 50; i++)
                {
                    for (int j = 0; j < 3; j++)
                    {
                        var e = new Entity { Description = "desc " + i + j, Size = j };
                        ctx.Entities.Add(e);
                    }
                }

                ctx.SaveChanges();
            }

            using (var ctx = new MyContext())
            {
                //ctx.Database.Log = Console.WriteLine;
                var batch1 = ctx.Entities.OrderBy(e => e.Nullable).Where(e => e.Size == 1).Select(x => new { x.Description, x.Id, x.Nullable, x.Size }).Skip(10).Take(10).ToList();
                var batch2 = ctx.Entities.OrderBy(e => e.Nullable).Where(e => e.Size == 1).Select(x => new { x.Description, x.Id, x.Nullable, x.Size }).Skip(20).Take(10).ToList();
            }
        }
    }

    public class MyContext : DbContext
    {
        public MyContext() : base("MyContext")
        {
            this.Database.Log = s => Log(s);
        }

        private static void Log(string s)
        {

        }

        public DbSet<Entity> Entities { get; set; }
    }

    public class Entity
    {
        public int Id { get; set; }
        public string Description { get; set; }
        public int Size { get; set; }
        public int? Nullable { get; set; }
    }
}
`

I hope you see the error with this one. I'm not sure if all of this is necessary

**Comment by maumar** [comment]
@luizfbicalho thanks for the follow up, with your changes I can reproduce the problem indeed.

**Comment by maumar** [comment]
Fixed in 8caee28305de907a23f766586e4a358299c3633c
